python/pyspark/sql/functions.py

```diff
@@ -1949,6 +1949,14 @@ def _create_judf(self):
         return judf
 
     def __call__(self, *cols):
+        for c in cols:
+            if not isinstance(c, (Column, str)):
```

Comment:
@HyukjinKwon Sorry for a delayed response, I am seldom online these days. You're right, it looks like an issue. I'll take a look at this, when I have more time